### DISCUSSION TITLE
Batch convert Gifs to Spritesheets (+ Batch convert folder of individual png to Spritesheet?)

### DISCUSSION THREAD

**Discussion by Lyskia37** [ImageMagick/ImageMagick#4744] [Help]
Hey, i have no experience at all using ImageMagick, batch and command line tools in general but after trying many things with zero results i need help :

I would like to make a 2 .bat files 

**1**. The most important one.
I would like to make a script that would allow me to **batch convert a lot of gifs into spritesheets.**

I found this .zsh script that seem to do that, but after installing bash and zsh on my Windows, i still don't manage to run it on Windows (it's meant to be used on Mac or Linux), also even if i would manage to get it work, i'm not sure if it covers my needs. 
I link it just in case it helps someone else : https://ddr0.ca/blog-posts/04.GIFs_To_PNGs

In a ideal world (additional features/option of the script that would be nice to have but non mandatory at all)
**a.** The name of the generated .png spritesheet would be "nameOfTheOriginalGif_AxB" (where A is the number of columns of the spritesheet and B is the number of rows of the spritesheet)
**b.** An option to remove all duplicate frames of the gif before it's converted to spritesheet (to avoid to have unnecessary double frames in the spritesheet)
**c.** possibility to choose one of those 3 options for the generated spritesheets
- Stack Horizontally : all the frames are combined in a single row .png (with the name "spritesheet_Ax1" - where A is the number of frames)
- Stack Vertically : all the frames are combined in a single column .png (with the name "spritesheet_1xB" - where B is the number of frames)
- Optimize space : depending on the dimensions of the gifs, the columns and rows numbers are optimized to make the spritesheet as "square" as possible. In game engines, it's better to have texture as condensed as possible. For example if the dimension of my gif is 10x30 pixels, and i have 12 frames, the most optimized optimized spritesheet would be 6 columns and 2 rows.

**2**. I would love to have an other script that would allow me to **batch convert a lot of folders containing .png frames into spritesheets**.
Like i choose a parent folder containing a bunch of child folders (each of them corresponding to a .png frames suite), and it generates a spritesheet for each child folder with the name "nameOfTheChildFolder_AxB" (where A is the number of columns of the spritesheet and B is the number of rows of the spritesheet)

In an ideal world, the a. b. c. additional features would also be available for this second script. 

Thanks a lot for your help 🙏

**Comment by snibgo** [comment]
> ... no experience at all using ImageMagick, batch and command line tools in general ...

Hmm, okay. No experience with bash either, I suppose. Do you want to gain this experience, or do you want someone to create the ideal script for you? I don't have enough free time for the latter.

As general suggestions:

You don't say what a "spritesheet" is, and nor does your link. I suppose it is a single-image montage of the input GIFs. Perhaps the input GIFs are all the same size.

If you are stacking horzontally or vertically, then in your ...

> "nameOfTheOriginalGif_AxB" (where A is the number of columns of the spritesheet and B is the number of rows of the spritesheet)

... A or B is simply a count of the inputs.

To optimize space, I suggest a calculation is needed to provide parameters for a "-tile" setting. This also gives the values of A and B.

So a script could be written to do that.

For "batch convert a lot of folders", I suggest another script that simply loops through subdirectories, calling the first script for each subdirectory. In Windows, I would use the "for" command.
